operator/pkg/rest/getter.go

```diff
@@ -38,7 +50,10 @@ func (c *ManifestRESTClientGetter) ToDiscoveryClient() (discovery.CachedDiscover
 	if err != nil {
 		return nil, err
 	}
-	return memory.NewMemCacheClient(discoveryClient), nil
+
+	c.client = memory.NewMemCacheClient(discoveryClient)
```

Comment:
caching a `MemCacheClient` with `discoveryClient` is never a good idea. Since the GV mappings will become static and never update. 
For e.g. on chart resources update for the same Manifest, with some new GVRs will fail from HelmClient since corresponding rest mapping are missing in `memCacheClient` (requests delegated to `discoveryClient`)

Alternative on my mind: listen for Manifest spec updates and each time invalidate `memCacheClient.Invalidate()` to that mapping are re-built. It might impact performance but cannot think of a better solution right now.

Reply:
e.g. https://github.com/helm/helm/blob/main/pkg/action/action.go#L248

Reply:
This is not true imo. memory.NewMemCacheClient returns `CachedDiscoveryInterface`.

This interface is described as:
```
// CachedDiscoveryInterface is a DiscoveryInterface with cache invalidation and freshness.
// Note that If the ServerResourcesForGroupVersion method returns a cache miss
// error, the user needs to explicitly call Invalidate to clear the cache,
// otherwise the same cache miss error will be returned next time.
```

Invalidate is called currently for all GVRs within `getClusterResources` which calls `GetTargetResources` which calls `transformManifestResources` which calls `convertToInfo` which calls `assignRestMapping` which calls `mapper.Reset()` if the mapping does not resist which calls `Invalidate` on the Mapping. Quite convoluted but I think this will work fine :D